### ISSUE TITLE
Pyright Not Reporting reportMissingModuleSource Error for typing_extensions Imports

### ISSUE THREAD

**Issue by StellaHuang95** [microsoft/pyright#7318]
**Describe the bug**
The `typing_extensions` library is a third-party library that requires user installation. Pyright seems to treat it as part of the standard library because it's included in the `typeshed` repository's `stdlib` folder. As a result, Pyright won't report a `reportMissingModuleSource` error if a user imports `typing_extensions` but lacks the corresponding source file. 

**Code or Screenshots**

```python
from typing_extensions import TypeAlias

a: TypeAlias = int
```
Pick an interpreter that doesn't have `typing_extensions` installed to run it.

**Expected Behavior:**
Pyright identifies `typing_extensions` as a third-party library and report a `reportMissingModuleSource` error if the corresponding source file is not available when the library is imported.

**Actual Behavior:**
Pyright does not report a `reportMissingModuleSource` error when `typing_extensions` is imported without the source file. However, running the code results in a runtime error:
```
Traceback (most recent call last):
  File "c:\Users\stellahuang\source\projects\folder1\file1.py", line 1, in <module>
    from typing_extensions import TypeAlias
ModuleNotFoundError: No module named 'typing_extensions'
```

**Comment by erictraut** [comment]
Yeah, `typing_extensions.pyi` is part of the stdlib typeshed stubs, but it's not technically part of stdlib. We need to live with this unfortunate side effect or add some hard-coded special-case logic in pyright. I'm not sure if this is important enough to justify a special case.

@StellaHuang95, can you provide more details about how this was found? Was it reported by a user? How important does the pylance team consider this issue?

**Comment by StellaHuang95** [comment]
This issue was initially brought up by a user who complained about [Quick fix suggests importing symbol that is not available in pythonVersion](https://github.com/microsoft/pylance-release/issues/4993). Bill is working on implementing a `pylance: create all stdlib indices` command that users can run to match their python version. But the concern here is that suggesting the `typing_extensions` in the quick fix could potentially lead to runtime errors if the user doesn't have it installed. One of the solutions we think of is to special case [here in _addMissingModuleSourceDiagnosticIfNeeded](https://github.com/microsoft/pyrx/blob/5430362e9386c659ca2b964cae5eed7b5a813890/packages/pyright/packages/pyright-internal/src/analyzer/checker.ts#L1723) to report errors for `typing_extensions`, which should be adequate for the issue.

The priority for this is P2. Also tagging @bschnurr in case I missed anything.

Other relevant issues that might be resolved by this fix:
https://github.com/microsoft/pylance-release/issues/5196
https://github.com/microsoft/pylance-release/issues/3785

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.352, which I just published.
